realm/realm-library/src/main/java/io/realm/internal/SharedRealm.java

```diff
@@ -161,10 +176,12 @@ long getNativePtr() {
 
     public void beginTransaction() {
         nativeBeginTransaction(nativePtr);
+        invokeListenerIfSchemaChanged();
     }
 
     public void commitTransaction() {
         nativeCommitTransaction(nativePtr);
+        invokeListenerIfSchemaChanged();
```

Comment:
Currently, both `Realm` and `DynamicRealm` can exist at the same time, then following code has problems:
```java
dynamicRealm.beginTransaction();
dynamicRealm.removeField("foo");
realm.getFoo(); // Column doesn't exist anymore
dynamicRealm.commitTransaction();
```

I am not quite sure if it worth to spend time on this problem since having both at the same doesn't seem be to a real requirement.
An easy fix is we just don't allow having them at the same time.
What do you think? @realm/java